File: Solutions/187. Repeated DNA Sequences/187. Repeated DNA Sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
        Set set=new HashSet(),res=new HashSet();
        for(int i=0;i+9<s.length();++i){
            String ten=s.substring(i,i+10);
            if(!set.add(ten))
                res.add(ten);
        }
        return new ArrayList(res);
    }
}

File: Solutions/187. Repeated DNA Sequences/187. Repeated DNA Sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        unordered_set<string> set,res;
        for(int i=0;i+9<s.size();++i){
            string ten=s.substr(i,10);
            if(set.find(ten)!=set.end()){
                res.insert(ten);
            }
            set.insert(ten);
        }
        return vector<string>(res.begin(),res.end());
    }
};